internal/io/http/client.go

```diff
@@ -350,20 +351,22 @@
 		}
 		return nil, nil, nil
 	case "body":
-		payload, err := decode(ctx, c)
+		payloads, err := decode(ctx, c)
 		if err != nil {
 			return nil, c, err
 		}
-		ro := &bodyResp{}
-		err = cast.MapToStruct(payload, ro)
-		if err != nil {
-			return nil, c, fmt.Errorf("invalid body response: %v", err)
-		}
 		if resp.StatusCode < 200 || resp.StatusCode > 299 {
```

Comment:
This was a bug, it actually wants to check ro.Code. Some http service do not use http status code, but put the code in the payload.

Reply:
updated.